### PR TITLE
[scnlib] update to 4.0.1

### PR DESCRIPTION
**Pull request by jiayuehua**
https://godbolt.org/z/EhsWsMvEK 

ports/scnlib/fix-SCN_HAS_STD_REGEX_MULTILINE-marco.patch is needed otherwise this port can't build successfully use gcc11.3, so add it back.

### CODE REVIEW

**Review by Mengna-Li — DISMISSED** [review]

File: ports/scnlib/fix-SCN_HAS_STD_REGEX_MULTILINE-marco.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+diff --git a/include/scn/fwd.h b/include/scn/fwd.h
+index 9b4892f..92f603b 100644
+--- a/include/scn/fwd.h
++++ b/include/scn/fwd.h
+@@ -740,10 +740,10 @@ SCN_GCC_POP
+ 
+ // Detect std::regex_constants::multiline:
+ // libc++ 15 and later, or libstdc++ 11.4 or later
+-// (2021-09-29 is the date of the commit introducing `multiline`,
```
Comment (data-queue) [inline ports/scnlib/fix-SCN_HAS_STD_REGEX_MULTILINE-marco.patch:9]: Can you submit this patch upstream?

Comment (jiayuehua) [reply ports/scnlib/fix-SCN_HAS_STD_REGEX_MULTILINE-marco.patch:9]: already submit to upstream https://github.com/eliaskosunen/scnlib/pull/130

**Comment by jiayuehua** [comment]
upstream has just merged the fix.

**Comment by jiayuehua** [comment]
@Mengna-Li

**Comment by ajtribick** [comment]
Upstream released this in version 3.0.2.

**Comment by Mengna-Li** [comment]
@jiayuehua Please update to 4.0.1. Thanks.

**Review by Mengna-Li — APPROVED** [review]

**Pull request merged**